server/sonar-server/src/main/java/org/sonar/server/measure/custom/ws/SearchAction.java

```diff
@@ -0,0 +1,180 @@
+  public static final String PARAM_PROJECT_KEY = "projectKey";
+
+  private final DbClient dbClient;
+  private final UserIndex userIndex;
+  private final CustomMeasureJsonWriter customMeasureJsonWriter;
+
+  public SearchAction(DbClient dbClient, UserIndex userIndex, CustomMeasureJsonWriter customMeasureJsonWriter) {
+    this.dbClient = dbClient;
+    this.userIndex = userIndex;
+    this.customMeasureJsonWriter = customMeasureJsonWriter;
+  }
+
+  @Override
+  public void define(WebService.NewController context) {
+    WebService.NewAction action = context.createAction(ACTION)
+      .setDescription("List custom measures. The project id or project key must be provided.")
+      .setSince("5.2")
+      .addFieldsParam(CustomMeasureJsonWriter.OPTIONAL_FIELDS)
+      .addPagingParams(100)
+      .setResponseExample(getClass().getResource("example-search.json"))
+      .setHandler(this);
+
+    action.createParam(PARAM_PROJECT_ID)
+      .setDescription("Project id")
+      .setExampleValue("ce4c03d6-430f-40a9-b777-ad877c00aa4d");
+
+    action.createParam(PARAM_PROJECT_KEY)
+      .setDescription("Project key")
+      .setExampleValue("org.apache.hbas:hbase");
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    String projectUuid = request.param(PARAM_PROJECT_ID);
+    String projectKey = request.param(PARAM_PROJECT_KEY);
+    List<String> fieldsToReturn = request.paramAsStrings(WebService.Param.FIELDS);
+    SearchOptions searchOptions = new SearchOptions()
+      .setPage(request.mandatoryParamAsInt(WebService.Param.PAGE),
+        request.mandatoryParamAsInt(WebService.Param.PAGE_SIZE));
+
+    DbSession dbSession = dbClient.openSession(false);
+    try {
+      ComponentDto project = searchProject(dbSession, projectUuid, projectKey);
+      List<CustomMeasureDto> customMeasures = searchCustomMeasures(dbSession, project, searchOptions);
+      int nbCustomMeasures = countTotalOfCustomMeasures(dbSession, project);
+      Map<String, User> usersByLogin = usersByLogin(customMeasures);
+      Map<Integer, MetricDto> metricsById = metricsById(dbSession, customMeasures);
+
+      writeResponse(response, customMeasures, nbCustomMeasures, project, metricsById, usersByLogin, searchOptions, fieldsToReturn);
+    } finally {
+      MyBatis.closeQuietly(dbSession);
+    }
+  }
+
+  private int countTotalOfCustomMeasures(DbSession dbSession, ComponentDto project) {
+    return dbClient.customMeasureDao().countByComponentUuid(dbSession, project.uuid());
+  }
+
+  private List<CustomMeasureDto> searchCustomMeasures(DbSession dbSession, ComponentDto project, SearchOptions searchOptions) {
+    return dbClient.customMeasureDao().selectByComponentUuid(dbSession, project.uuid(), searchOptions);
+  }
+
+  private void writeResponse(Response response, List<CustomMeasureDto> customMeasures, int nbCustomMeasures, ComponentDto project, Map<Integer, MetricDto> metricsById,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method has 8 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)